File: Core/Model/Transaction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace CampBank.Core.Model
{
    public class Transaction
    {
        public int Id { get; set; }
        public TransactionType Type { get; set; }
        public float Amount { get; set; }
        public int KidId { get; set; }
        public Kid Kid { get; set; }
        [Required]
        public string UserName { get; set; }
        public DateTime TimeStamp { get; set; }
    }
}

File: ClientApp/app/models/transaction.ts
import { TransactionType } from "./transactionType";

export interface Transaction {
    id: number;
    type: TransactionType;
    amount: number;
    kidId: number;
    userName: string;
    timeStamp: Date;
}

export interface SaveTransaction {
    type: TransactionType;
    amount: number;
    kidId: number;
}